server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeQProfileMeasureStep.java

```diff
@@ -74,19 +77,36 @@ public QProfiles createForAny(Component component) {
     }
 
     @Override
+    public void visitFile(Component file, Path<QProfiles> path) {
+      String languageKey = file.getFileAttributes().getLanguageKey();
+      if (languageKey == null) {
+        // No qprofile for unknown languages
+        return;
+      }
+      if (!analysisMetadataHolder.getQProfilesByLanguage().containsKey(languageKey)) {
+        // Should never occurs except if scanner report is buggy (ie list of qprofile is not consistent with file languages)
+        // TODO Should we fail fast, log?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1135)